### PR TITLE
feat: set maximum and minimum timestamp to get electricity data

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#402

### CODE REVIEW

File: client/app/scripts/app.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ angular
     'angular-nw-weather'
   ])
   .config(function ($stateProvider, $urlRouterProvider, $httpProvider, cfpLoadingBarProvider) {
+    $httpProvider.defaults.cache = true;
```
Comment (amitaibu) [inline client/app/scripts/app.js]: remove

File: client/app/scripts/services/chartUsagePeriods.js [hunk 2]
```diff
@@ -4,44 +4,91 @@ angular.module('negawattClientApp')
...
+        if (angular.isDefined(period.next) && angular.isDefined(period.previous)) {
+          // Comming from the calculation.
+          timestamp.next = period.next;
+          timestamp.previous = period.previous;
+        }
+      },
+      getPrevious: function() {
+        return moment.unix(this.next).subtract(this.config.chart_default_time_frame, this.config.frequency).unix();
+      },
+      getNext: function() {
```
Comment (amitaibu) [inline client/app/scripts/services/chartUsagePeriods.js]: remove

File: client/app/scripts/services/chartUsagePeriods.js [hunk 3]
```diff
@@ -4,44 +4,91 @@ angular.module('negawattClientApp')
...
+      timestamp.min = +limits.min;
+    }
 
-      // Extend period object with new values.
-      extend(times, {
-        previous: filters['filter[timestamp][value][0]'],
-        next: filters['filter[timestamp][value][1]']
-      });
+    /**
+     * Extend the time stamp object with thr actual peiod selection.
```
Comment (amitaibu) [inline client/app/scripts/services/chartUsagePeriods.js]: thr =>the

File: client/app/scripts/services/meter.js [hunk 4]
```diff
@@ -46,6 +46,13 @@ angular.module('negawattClientApp')
     };
 
     /**
+     * Return the electricity time interval according the meters requested.
+     */
+    this.getElectricityInterval = function() {
+      return cache.electricity_time_interval;
```
Comment (amitaibu) [inline client/app/scripts/services/meter.js]: ``return this.get().electricity_time_interval``

File: negawatt/modules/custom/negawatt_restful/plugins/formatter/meters_total/NegawattFormatterMetersTotal.class.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      $query->join('node', 'n', 'n.nid = e.meter_nid');
+      $query->join('og_membership', 'og', 'og.etid = n.nid');
+      $query->condition('og.entity_type', 'node');
+      $query->condition('og.gid', $filter['account']);
+      unset($filter['account']);
+    }
+
+    // Make sure we handled all the filter fields.
+    if (!empty($filter)) {
+      throw new \Exception('NegawattFormatterMetersTotal::prepare: Unknown fields in filter.');
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/formatter/meters_total/NegawattFormatterMetersTotal.class.php]: => 'Unknown fields in filter.'

File: negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/1.0/meter__1_0.inc [hunk 6]
```diff
@@ -12,4 +12,6 @@ $plugin = array(
     'satec_meter' => 'satec_meters',
   ),
   'authentication_types' => TRUE,
-);
+  // Add totals to output.
+  'formatter' => 'meters_total',
+  );
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/1.0/meter__1_0.inc]: indentation error

File: client/app/scripts/app.js (hunk 1 above)
Comment (ceoaliongroo) [inline client/app/scripts/app.js]: removed.

**Comment by ceoaliongroo** [comment]
Merged.

**Pull request closed without merging**